eZ/Publish/Core/Persistence/Cache/UrlAliasHandler.php

```diff
@@ -62,11 +65,30 @@ public function publishUrlAliasForLocation(
             $updatePathIdentificationString
         );
 
-        $this->cache->invalidateTags([
-            $this->cacheIdentifierGenerator->generateTag(self::URL_ALIAS_LOCATION_IDENTIFIER, [$locationId]),
-            $this->cacheIdentifierGenerator->generateTag(self::URL_ALIAS_LOCATION_PATH_IDENTIFIER, [$locationId]),
-            $this->cacheIdentifierGenerator->generateTag(self::URL_ALIAS_NOT_FOUND_IDENTIFIER),
-        ]);
+        try {
+            $existingLocationAliases = $urlAliasHandler->listURLAliasesForLocation($locationId);
+        } catch (BadStateException $e) {
+            $existingLocationAliases = [];
+        }
+
+        $existingLocationAliasesTags = [];
+        foreach ($existingLocationAliases ?? [] as $existingAlias) {
+            $existingLocationAliasesTags[] = $this->cacheIdentifierGenerator->generateKey(
```

Comment:
Shouldn't this be:
```suggestion
            $existingLocationAliasesTags[] = $this->cacheIdentifierGenerator->generateTag(
```
?

The keys are tagged with this tag anyway via `\eZ\Publish\Core\Persistence\Cache\UrlAliasHandler::getCacheTags` so invalidating a tag instead of a key should be enough. Especially that you later on call `invalidateTags` instead of `deleteItem($keys)`.